api/src/main/java/org/apache/iceberg/encryption/EncryptedOutputFile.java

```diff
@@ -37,8 +39,15 @@
   OutputFile encryptingOutputFile();
 
   /**
-   * Metadata about the encryption key that is being used to encrypt the associated
+   * Metadata about the encryption keys and other crypto parameters used to encrypt the associated
    * {@link #encryptingOutputFile()}.
    */
   EncryptionKeyMetadata keyMetadata();
+
+  /**
+   * Parameters of native encryption (if used for this file)
+   */
+  default NativeFileCryptoParameters nativeEncryptionParameters() {
```

Comment:
Can we reuse `keyMetadata ` instead of introducing `nativeEncryptionParameters `? Here are reasons:
1. Both `keyMetadata` and `nativeEncryptionParameters` are used for keys and crypto parameters.
2. Each file will have either one, not both. It depends whether streaming encryption or native encryption is used for this file.

NativeFileCryptoParameters can implement interface `EncryptionKeyMetadata` in that sense.

Reply:
Not quite; all encrypted files (inc natively encrypted) have `key_metadata`, since it is required to decrypt them. But, as I mentioned in the previous comment, I'll play with applying a new interface to the output too, so maybe the `EncryptedOutputFile` doesn't need to change.